orttraining/orttraining/core/framework/gradient_graph_builder.h

```diff
@@ -22,7 +23,13 @@
 namespace onnxruntime {
 namespace training {
 
-typedef std::unordered_set<const Node*> NodeSet;
+// typedef std::unordered_set<const Node*> NodeUnorderedSet;
+struct node_compare {
+    bool operator() (const Node* lhs, const Node* rhs) const {
+        return lhs->Name() < rhs->Name();
+    }
+};
+typedef std::set<const Node*, node_compare> NodeSet;
```

Comment:
>NodeSet [](start = 44, length = 7)

please add a comment about why this affects gradient graph. Plus, there are some other undered_set/unordered_map in here, like x/y_node_arg_names. Should that change to set too?